### PR TITLE
Migrate from Circle CI to GitHub Actions

### PR DESCRIPTION
**Pull request by colincasey**
[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

### CODE REVIEW

File: integration-test/openjdk-11/exception/test.sh [hunk 1]
```diff
@@ -8,16 +8,16 @@ function_invocation_response_file="$(mktemp)"
 port=54321
 
 pushd "${test_dir}"
-./mvnw compile
+./mvnw --no-transfer-progress compile
```
Comment (Malax) [inline integration-test/openjdk-11/exception/test.sh:11]: We probably want `--batch` here, which includes `--no-transfer-progress`.

File: pom.xml [hunk 2]
```diff
@@ -290,6 +290,7 @@
                 <version>3.4.1</version>
                 <configuration>
                     <source>${maven.compiler.source}</source>
+                    <doclint>all,-syntax</doclint>
```
Comment (Malax) [inline pom.xml:293]: Is this the same issue as with the SDK where `javadoc` picks up an annotation as a doctag?

**Review by Malax — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (colincasey) [reply pom.xml:293]: yes.  and i can't figure out why or reproduce it 😞

File: pom.xml [hunk 3]
```diff
@@ -288,9 +288,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.4.1</version>
-                <configuration>
-                    <source>${maven.compiler.source}</source>
-                </configuration>
```
Comment (Malax) [inline pom.xml:293]: I assume this was removed because it's the default anyway?

**Review by Malax — APPROVED** [review]
Thanks for taking care of this! Appreciated! 👍🏻

**Pull request merged**